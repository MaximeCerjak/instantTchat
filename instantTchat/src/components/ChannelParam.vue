<template>
    <div class="sidebar-channel-param">
        <div class="handleChannel">
            <h2>{{ channel.name }}</h2>
            <button @click="deleteChannel">Supprimer le canal</button>
            <button @click="sendInvitation">Inviter des membres</button>
        </div>
        <div class="users-list">
            <h2>Users</h2>
            <ul>
                <li v-for="member in members" :key="member.id">
                    {{ member }}
                </li>
            </ul>
        </div>
    </div>
</template>

<script setup>
import { ref } from 'vue';

const properties = defineProps({
    channel: {
        type: Object,
        required: true
    },
    users: {
        type: Array,
        required: true
    },
    messages: {
        type: Array,
        required: true
    }
});

const channel = ref(properties.channel);
const members = ref(properties.users);
const messages = ref(properties.messages);

console.log(channel.value);
console.log(members.value);
console.log(messages.value);

</script>

<style scoped>

.sidebar-channel-param {
    display:flex;
    flex-direction: column;
    position: absolute;
    top: 8.7%;
    right: 0;
    height: 100%;
    background-color: #181818;
    width: 15vw;
    color: rgb(175, 175, 175);
}

.handleChannel {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
}

.handleChannel button {
    height: 50px;
    margin: 10px 0;
    border-radius: 10px;
    background-color: #59595966;
    border: none;
    outline-style: none;
    color: white;
    font-size: 1rem;
    margin-right: 2px;
    cursor: pointer;
    padding: 1rem;
}

.handleChannel h2 {
    font-size: 1.2rem;
    margin-bottom: 10px;
}

.users-list {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    background-color: #000000;
}

</style>