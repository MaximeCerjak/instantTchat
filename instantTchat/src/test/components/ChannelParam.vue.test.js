import {expect,describe,it,beforeEach} from "vitest";
import { mount } from '@vue/test-utils';
import { createPinia } from 'pinia';
import ChannelParam from '../../components/ChannelParam.vue';

describe('ChannelParam', () => {
    let wrapper;
    const pinia = createPinia();

    beforeEach(() => {
        wrapper = mount(ChannelParam, {
            global: {
                plugins: [pinia],
            },
            props: {
                channel: {
                    name: 'channel1',
                    creator: 'user1',
                    id: 1,
                    users: [],
                    messages: [],
                },
            },
        });
    });

    it('should render the component correctly', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

});
