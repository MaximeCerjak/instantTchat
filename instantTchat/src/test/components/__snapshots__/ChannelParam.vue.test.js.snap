// Vitest Snapshot v1

exports[`ChannelParam > should render the component correctly 1`] = `
"<div data-v-e856487c=\\"\\" class=\\"sidebar-channel-param\\">
  <div data-v-e856487c=\\"\\" class=\\"handleChannel\\">
    <h2 data-v-e856487c=\\"\\">user1</h2><button data-v-e856487c=\\"\\">Supprimer le canal</button><button data-v-e856487c=\\"\\">Inviter des membres</button>
  </div>
  <div data-v-e856487c=\\"\\" class=\\"users-list\\">
    <h2 data-v-e856487c=\\"\\">Users</h2>
    <ul data-v-e856487c=\\"\\"></ul>
  </div>
</div>
<!--v-if-->
<!--v-if-->
<!--v-if-->"
`;
